src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY_TYPE = "supply,";
+    private static final String BUY_TYPE = "buy,";
+    private static final String RESULT_TYPE = "result,";
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final int MINUS_ONE_INDEX = -1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String readData = readFromFile(fromFileName);
+        StringBuilder dataToWrite = new StringBuilder();
+        int supplySum = getCalculations(readData, SUPPLY_TYPE);
+        int buySum = getCalculations(readData, BUY_TYPE);
+        int result = supplySum - buySum;
+
+        dataToWrite.append(SUPPLY_TYPE).append(supplySum).append(NEW_LINE)
+                .append(BUY_TYPE).append(buySum).append(NEW_LINE)
+                .append(RESULT_TYPE).append(result);
+
+        writeToFile(toFileName, dataToWrite.toString());
+    }
+
+    public String readFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
bad naming, don't name after data type or value, name after role it's supposed to server, what it represents